File: pset1/exercises/cash.c
#include <stdio.h>
#include <cs50.h>

float change(void);

void coins (float n);

int main(void)
{
    float n = change();
    coins(n);
}

float change(void)
{
    float n;
    do
    {
        n = get_float("Right change: ");
    }
    while(n < 0);
    return n;
}

void coins (float n)
{
    int coin = (int) (n * 100);
    int coins = 0;
    while(coin - 25 >= 0)
    {
        coin -= 25;
        coins++;
    }
    while(coin - 10 >= 0)
    {
        coin -= 10;
        coins++;
    }
    while(coin - 5 >= 0)
    {
        coin -= 5;
        coins++;
    }
    while(coin - 1 >= 0)
    {
        coin -= 1;
        coins++;
    }
    printf("%i\n", coins);
}

File: pset6/exercises/cash.py
from cs50 import get_float

def mony():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            break
    return n

def coins(n):
    n = int(n *100)
    coins = 0
    while (n - 25) >= 0:
        coins += 1
        n -= 25
    while (n - 10) >= 0:
        coins += 1
        n -= 10
    while (n - 5) >= 0:
        coins += 1
        n -= 5
    while (n - 1) >= 0:
        coins += 1
        n -= 1
    print(coins)

mony = mony()
coins(mony)